compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/generic_generation.rs

```diff
@@ -0,0 +1,169 @@
+pub struct PartitionDefinition {
+    pub precision: Precision,
+    pub norm2: f64,
+}
+
+impl PartialOrd for PartitionDefinition {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match self.precision.cmp(&other.precision) {
+            std::cmp::Ordering::Equal => self.norm2.partial_cmp(&other.norm2),
+            ordering => Some(ordering),
+        }
+    }
+}
+
+pub fn generate_generic_parameters(
+    partitions: Vec<PartitionDefinition>,
+    generate_fks: bool,
+) -> CircuitKeys {
+    let mut dag = unparametrized::Dag::new();
+
+    for def_a in partitions.iter() {
+        for def_b in partitions.iter() {
+            if def_a == def_b {
+                continue;
+            }
+            let inp_a = dag.add_input(def_a.precision, Shape::number());
+            let lut_a = dag.add_lut(inp_a, FunctionTable::UNKWOWN, def_a.precision);
+            let _weighted_a = dag.add_linear_noise(
+                [lut_a],
+                LevelledComplexity::ZERO,
+                [def_a.norm2.sqrt()],
+                Shape::number(),
+                "",
+            );
+
+            let inp_b = dag.add_input(def_b.precision, Shape::number());
+            let lut_b = dag.add_lut(inp_b, FunctionTable::UNKWOWN, def_b.precision);
+            let weighted_b = dag.add_linear_noise(
+                [lut_b],
+                LevelledComplexity::ZERO,
+                [def_b.norm2.sqrt()],
+                Shape::number(),
+                "",
+            );
+
+            dag.add_composition(weighted_b, inp_a);
+
+            if generate_fks && def_a > def_b {
+                let inp_a = dag.add_input(def_a.precision, Shape::number());
+                let lut_a = dag.add_lut(inp_a, FunctionTable::UNKWOWN, def_a.precision);
+                let _weighted_a = dag.add_linear_noise(
+                    [lut_a],
+                    LevelledComplexity::ZERO,
+                    [def_a.norm2.sqrt()],
+                    Shape::number(),
+                    "",
+                );
+
+                let inp_b = dag.add_input(def_b.precision, Shape::number());
+                let lut_b = dag.add_lut(inp_b, FunctionTable::UNKWOWN, def_b.precision);
+                let _weighted_b = dag.add_linear_noise(
+                    [lut_b],
+                    LevelledComplexity::ZERO,
+                    [def_b.norm2.sqrt()],
+                    Shape::number(),
+                    "",
+                );
+
+                let _ = dag.add_linear_noise(
+                    [lut_a, lut_b],
+                    LevelledComplexity::ZERO,
+                    [0., 0.],
+                    Shape::number(),
+                    "",
+                );
+            }
+        }
+    }
+
+    let precisions: Vec<_> = partitions.iter().map(|def| def.precision).collect();
+    let n_partitions = precisions.len();
+    let p_cut = PartitionCut::maximal_partitionning(&dag);
+    let config = Config {
```

Comment:
optimizer config should be passed like optimize_multi function to be customisable

Reply:
Btw seeing optimize_multi I see we always use CpuComplexity for the complexity model (may open an issue for that)

Reply:
Done, and I created an issue.